modules/queue/queue_wrapped.go

```diff
@@ -138,21 +139,68 @@ func (q *WrappedQueue) Name() string {
 
 // Push will push the data to the internal channel checking it against the exemplar
 func (q *WrappedQueue) Push(data Data) error {
-	if q.exemplar != nil {
-		// Assert data is of same type as r.exemplar
-		value := reflect.ValueOf(data)
-		t := value.Type()
-		exemplarType := reflect.ValueOf(q.exemplar).Type()
-		if !t.AssignableTo(exemplarType) || data == nil {
-			return fmt.Errorf("Unable to assign data: %v to same type as exemplar: %v in %s", data, q.exemplar, q.name)
-		}
+	if !assignableTo(data, q.exemplar) {
+		return fmt.Errorf("unable to assign data: %v to same type as exemplar: %v in %s", data, q.exemplar, q.name)
 	}
+	atomic.AddInt64(&q.numInQueue, 1)
 	q.channel <- data
 	return nil
 }
 
+func (q *WrappedQueue) flushInternal(timeout time.Duration) error {
+	q.lock.Lock()
+	if q.internal == nil {
+		q.lock.Unlock()
+		return fmt.Errorf("not ready to flush wrapped queue %s yet", q.Name())
```

Comment:
Should this really return an error? A hairy problem indeed.... 🤔

Reply:
the queue isn't flushed - so I think it should... This function isn't actually wired into anything though as wrapped queues don't actually implement Flushable - (they almost do).

I probably need to rethink this.

Reply:
Maybe it's OK to return an error under the following light: the admin wants to know that everything was processed, nothing is pending, but the worker is still initializing, so guess what? _I don't know if anything is pending_, hence I return an error.

Reply:
yeah that's the interpretation I mean.